src/shadowbox/server/run_action.sh

```diff
@@ -14,20 +14,21 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-echo "** Currently broken, use yarn do shadowbox/docker/run instead **" >&2
-exit 1
-
 do_action shadowbox/server/build
 
 export LOG_LEVEL="${LOG_LEVEL:-debug}"
 export SB_PUBLIC_IP="${SB_PUBLIC_IP:-$(curl https://ipinfo.io/ip)}"
 # WARNING: The SB_API_PREFIX should be kept secret!
 export SB_API_PREFIX=TestApiPrefix
 export SB_METRICS_URL=https://metrics-test.uproxy.org
-export SB_STATE_DIR=/tmp/outline
+export SB_OUTLINE_DIR=${SB_OUTLINE_DIR:-$(mktemp -d)}
+mkdir -p ${SB_OUTLINE_DIR}/bin ${SB_OUTLINE_DIR}/persisted-state
+
+cp third_party/prometheus/prometheus ${SB_OUTLINE_DIR}/bin/prometheus
```

Comment:
This is effectively duplicating the Dockerfile. If we move things around it's going to break, and it's also not consistent, so you may be testing something different.

Here is a suggestion: move the creation of the bin directory part of the build script. Then all the Dockerfile needs to do is COPY the bin dir. No need for duplication the construction logic.